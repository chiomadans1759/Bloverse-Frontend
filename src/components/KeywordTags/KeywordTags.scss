#keyword-tags {
  .card {
    width: auto;
    height: 1.938rem;
    margin-right: 0.5rem;
    margin-bottom: 0.5rem;
  }
  .svg-inline--fa {
    cursor: pointer;
    font-size: 0.8rem;
    margin-top: 0.125rem;
  }
  .item-content-block {
    width: 17.125rem;
    display: flex;
    flex-wrap: wrap;
  }
  width: 100%;
  .max {
    color: #B7B7B7;
    font-size: 0.75rem;
    margin: 0;
  }
  .text {
    background: transparent;
    border: 0.063rem solid rgba(0, 0, 0, 0.15);
    height: 1.938rem;
    width: 1.938rem;
    color: #096DD9;
    border-radius: 50% !important;
  }
  p {
    display: inline-block;
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
  .close {
    margin-left: 0.3rem;
    &:last-child {
      margin-left: 1rem;
    }
  }
}