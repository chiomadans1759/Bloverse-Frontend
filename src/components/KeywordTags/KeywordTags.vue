<template>
  <main id="keyword-tags">
    <div class="item-content-block d-flex">
      <div class="card" v-for="(keyword, index) in keyWordLinks" :key="index">
        <div class="card-content p-1">
          <div>
             <span class="keyword">Keyword</span>
             <span class="closeIcon" @click.prevent="removeItem(index)">
              <i class="fal fa-times d-flex justify-content-end close"></i>
             </span>
          </div>
        </div>
      </div>
       <span class="morekeys" @click.prevent="addKeyWord()" v-if="addWord">
          <button class="btn rounded-btn-circle d-flex justify-content-center ml-3 text">
            <i class="fal fa-plus"></i>
          </button>
       </span>
  </div>
    <p class="d-flex justify-content-end max">Maxinum of 5</p>
  </main>
</template>

<script>
export default {
  data(){
    return{
      addWord: true,
      keyWordLinks:[
        {
          keyword:''
        }
      ],
    }
  },
  methods: {
    removeItem(index) {
      this.keyWordLinks.splice(index,1);
      if(this.keyWordLinks.length == 5) {
        this.addWord = false;
      } else {
        this.addWord = true;
      }
    },
    addKeyWord() {
      this.keyWordLinks.push({
        keyword:''
      })
      if(this.keyWordLinks.length == 5) {
        this.addWord = false;
      } else {
        this.addWord = true;
      }
    }
  }
}
</script>

<style lang="scss" scoped>
  @import './KeywordTags';
</style>

