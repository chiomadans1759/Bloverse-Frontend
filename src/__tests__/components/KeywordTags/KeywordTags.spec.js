import { mount, RouterLinkStub } from '@vue/test-utils';
import KeywordTags from '@/components/KeywordTags/KeywordTags.vue';

describe('KeywordTags Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(KeywordTags, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should close socialTag', () => {
    const wrapper = mount(KeywordTags, {
      stubs: { RouterLink: RouterLinkStub }
    });
    wrapper.find('.closeIcon').trigger('click')
  })
})
