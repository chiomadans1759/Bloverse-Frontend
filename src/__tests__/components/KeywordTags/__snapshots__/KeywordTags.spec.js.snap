// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordTags Component should render correctly 1`] = `
<main
  id="keyword-tags"
>
  <div
    class="item-content-block d-flex"
  >
    <div
      class="card"
    >
      <div
        class="card-content p-1"
      >
        <div>
          <span
            class="keyword"
          >
            Keyword
          </span>
           
          <span
            class="closeIcon"
          >
            <i
              class="fal fa-times d-flex justify-content-end close"
            />
          </span>
        </div>
      </div>
    </div>
     
    <span
      class="morekeys"
    >
      <button
        class="btn rounded-btn-circle d-flex justify-content-center ml-3 text"
      >
        <i
          class="fal fa-plus"
        />
      </button>
    </span>
  </div>
   
  <p
    class="d-flex justify-content-end max"
  >
    Maxinum of 5
  </p>
</main>
`;
